corehq/apps/case_search/const.py

```diff
@@ -1,10 +1,59 @@
 from __future__ import absolute_import
 from __future__ import unicode_literals
 
+from collections import namedtuple
 
+# Case properties nested documents
 CASE_PROPERTIES_PATH = 'case_properties'
+VALUE = 'value'
+
+# Case indices nested documents
 INDICES_PATH = 'indices'
 REFERENCED_ID = 'referenced_id'
 IDENTIFIER = 'identifier'
-VALUE = 'value'
+
+# Added to each case response when case searches are performed
 RELEVANCE_SCORE = "commcare_search_score"
+
+# Added to each case on the index for debugging when a case was added to ES
+INDEXED_ON = '@indexed_on'
+
+# Properties added to the case search mapping to provide extra information
+SYSTEM_PROPERTIES = [
+    CASE_PROPERTIES_PATH,
+    INDEXED_ON,
+]
+
+# Properties that are inconsitent between case models stored in HQ and casedb
+# expressions. We store these as case properties in the case search index so
+# they are easily searchable, then remove them when pulling the case source
+# from ES
+SpecialCaseProperty = namedtuple('SpecialCaseProperty', 'key value_getter sort_property')
```

Comment:
@proteusvacuum I'm curious why we took the approach of adding these to the index instead of transforming the query to search the already indexed fields?

Reply:
Honestly, I don't remember. What you are suggesting sounds like it could have been a better idea.